Automattic/wp-calypso
# Support cross origin URL in gridicons

## Proposed Changes

We're working on moving the Help Center from ETK to Jetpack. The easiest path was to build a Help Center bundle and publish it to widgets.wp.com. Then enqueue in Jetpack. All works great.

The only issue I noticed was that Gridicons were blocked. site.wordpress.com/wp-admin was trying to load them from widgets.wp.com. `use` elements block cross-origin URLs and [cannot be configured to do otherwise](https://developer.mozilla.org/en-US/docs/Web/SVG/Element/use#:~:text=CSS%20inheritance.-,For%20security%20reasons,-%2C%20browsers%20may%20apply). This PR adds a client-side proxy to handle that case. The proxy will only run for cross-origin SVGs.

## Why are these changes being made?
So Gridicons can run in all contexts.

## Testing Instructions
0. cd into `apps/help-center` and run `yarn dev --sync`.
1. Follow the Simple sites instructions of [this PR](https://github.com/Automattic/jetpack/pull/38093).
2. Open the Help Center.
3. Click "Still need help?".
4. Ask Wapuu something.
5. Under its answer, the Related Guides card should have an icon. Like so

![image](https://github.com/Automattic/wp-calypso/assets/17054134/dc3dd3e1-6d03-4a8c-98a4-e38051228bb9)
